### PR TITLE
Fetching otter-db remarks in lookup endpoint

### PR DESCRIPTION
**Pull request by mira13**
**Description**

Remarks added to lookup endpoint, for work on top otter database.

**Use case**

zmap-web transcription info window needs remarks

**Benefits**

possibility to view remarks in zmap-web

**Possible Drawbacks**

None

**Testing**

Have you *added/modified unit tests to test the changes?
No

If so, do the tests pass/fail?

**Have you run the entire test suite and no regression was detected?**
Yes

### CODE REVIEW

File: lib/EnsEMBL/REST/Model/Lookup.pm [hunk 1]
```diff
@@ -24,6 +24,7 @@ use namespace::autoclean;
 use Try::Tiny;
 use Catalyst::Exception;
 use Scalar::Util qw/weaken/;
+use Bio::EnsEMBL::DBSQL::AttributeAdaptor;
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/Lookup.pm]: Not required unless you instantiate your own AttributeAdaptor. get_adaptor takes care of the import for you

File: lib/EnsEMBL/REST/Model/Lookup.pm [hunk 2]
```diff
@@ -258,6 +259,19 @@ sub find_object_by_stable_id {
   return $self->find_object($id, $species, $object_type, $db_type);
 }
 
+sub fetch_remarks_by_transcript {
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/Lookup.pm]: A comment here would do wonders for us understanding why this exists.

File: lib/EnsEMBL/REST/Model/Lookup.pm [hunk 3]
```diff
@@ -258,6 +259,19 @@ sub find_object_by_stable_id {
...
+  my ($self, $object, $code) = @_;
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/Lookup.pm]: $code appears to be unused.

File: lib/EnsEMBL/REST/Model/Lookup.pm [hunk 4]
```diff
@@ -483,6 +497,7 @@ sub transcript_feature {
   my $transcript = $self->find_object($id, $species, 'Transcript', $db_type);
   $features = $self->features_as_hash($id, $species, 'Transcript', $db_type, $transcript);
   $features->{Exon} = $self->Exon($transcript, $species, $db_type);
+  $features->{Remarks} = $self->fetch_remarks_by_transcript($transcript);
```
[the remark this answers is not in the document]
Comment (nerdstrike) [reply lib/EnsEMBL/REST/Model/Lookup.pm]: I agree. This applies a performance penalty to all Transcript feature requests through the public REST server. Please take steps to mitigate this. Any new endpoint option will need to be documented as well.

File: lib/EnsEMBL/REST/Model/Lookup.pm [hunk 5]
```diff
@@ -258,6 +259,19 @@ sub find_object_by_stable_id {
...
 }
 
+sub fetch_remarks_by_transcript {
+  my ($self, $object, $code) = @_;
+  my $object_id;
+  if (defined($object)) {
+    $object_id = $object->dbID();
+  }
+  my $c = $self->context();
+  my $db_type = $c->request->param('db_type') || 'core';
```
Comment (mkszuba) [inline lib/EnsEMBL/REST/Model/Lookup.pm]: Logically speaking we should use // and not || here, shouldn't we? I mean, it isn't particularly like that we would like to call our database type "0" but in theory we _could_ do it.

**Comment by mkszuba** [comment]
Any news on implementing the changes, @mira13?

**Comment by markmcdowall** [comment]
Hi @mira13,

Have you had a chance to have a look at these changes?

Cheers

**Comment by tgrego** [comment]
about 3 months with no input.
closing

**Pull request closed without merging**